src/sdp_proxy/discovery_proxy.cpp

```diff
@@ -64,6 +71,25 @@ DiscoveryProxy::DiscoveryProxy(Ncp::ControllerOpenThread &aNcp, Mdns::Publisher
 {
 }
 
+void DiscoveryProxy::Init(void)
+{
+    otDnsQueryConfig config = {
+        .mServerSockAddr  = {.mAddress = {}, .mPort = 0},
+        .mResponseTimeout = 0,
+        .mMaxTxAttempts   = 0,
+        .mRecursionFlag   = OT_DNS_FLAG_RECURSION_DESIRED,
+        .mNat64Mode       = OT_DNS_NAT64_DISALLOW,
+        .mNetif           = OT_NETIF_BACKBONE,
+    };
+    in_addr upstreamDnsAddr;
+
+    inet_aton(OTBR_CONFIG_UPSTREAM_DNS, &upstreamDnsAddr);
```

Comment:
Shall we use `inet_pton()` instead in case OTBR_CONFIG_UPSTREAM_DNS is an IPv6 address?

Ref: https://stackoverflow.com/questions/5328070/how-to-convert-string-to-ip-address-and-vice-versa